helsingborg-stad/app-mitt-helsingborg
# Feat/summary list styling

## Feature description
Styles the summary list according to design, and adds the functionality of an edit-button.

## Solution description
Changes the GroupedList component to use the Fieldset component in the same was as its used in the EditableList, and adds the ability to toggle between editable and not. Changes the SummaryListItem to incorporate the new styling, and the non-editable mode, again very similar to EditableList. 

Also some minor changes to type the GroupedList component. 

## Possible refactoring
There's a bit of repeating ourselves going on here, and we should probably refactor a bit to reduce it. The GroupedList can be made into a more general list component, allowing for both grouping and not, and then the editable list and summary list can be two different ways of using it.

## How to test the feature?
Load the test form, and enter some things so that the summary list appears. Or use the storybook. 

## Is there any existing behaviour change of other features due to this code change?
No.

## Covered unit tests cases / E2E test cases?
No.

## Are your code structured in a way so that reviewers can understand it?
Yes.

## Was this feature tested in the following environments?
- [x] On a iOS device/simulator.
